src/lib/common/string.cpp

```diff
@@ -999,21 +999,30 @@ std::string double2string(double f)
 }
 
 
+
 /*****************************************************************************
 *
 * isodate2str -
 *
 * FIXME P6: change implementation to use gmtime_r
 *
 */
-std::string isodate2str(long long timestamp)
+std::string isodate2str(double timestamp)
 {
   // 80 bytes is enough to store any ISO8601 string safely
   // We use gmtime() to get UTC strings, otherwise we would use localtime()
-  // Date pattern: 1970-04-26T17:46:40.00Z
-  char   buffer[80];
-  time_t rawtime = (time_t) timestamp;
-  strftime(buffer, sizeof(buffer), "%Y-%m-%dT%H:%M:%S.00Z", gmtime(&rawtime));
+  // Date pattern: 1970-04-26T17:46:40.000Z
+
+  char    buffer[80];
+  time_t  seconds   = (time_t) timestamp;
+  double  ms        = timestamp - (double) seconds;
+  int     micros    = ms * 1000000;
+  int     millis    = (micros + 1) / 1000;   // (timestamp - seconds) * 1000 gives rounding errors ...
```

Comment:
Maybe a comment should be added explaining why the "+1" is needed? It can be a little tricky...

Reply:
Fixed in 4d8fe4857

Reply:
Fixed in 4d8fe48